xrpl/transaction/main.py

```diff
@@ -44,17 +55,32 @@ def safe_sign_transaction(transaction: Transaction, wallet: Wallet) -> Transacti
     Returns:
         The signed transaction blob.
     """
-    # Increment the wallet sequence number, since we're about to use one.
-    wallet.next_sequence_num += 1
-    transaction_json = transaction_json_to_binary_codec_form(transaction.to_dict())
-    transaction_json["SigningPubKey"] = wallet.public_key
+    transaction_json = _prepare_transaction(transaction, wallet)
     serialized_for_signing = encode_for_signing(transaction_json)
     serialized_bytes = bytes.fromhex(serialized_for_signing)
     signature = sign(serialized_bytes, wallet.private_key)
     transaction_json["TxnSignature"] = signature
     return cast(Transaction, Transaction.from_xrpl(transaction_json))
```

Comment:
in the future, we're going to want to have generalized ability to copy a model and change/add/remove specific fields on the copy.

Reply:
Added a backlog ticket